cpu/cpu_dragonfly.go

```diff
@@ -0,0 +1,173 @@
+package cpu
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"reflect"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/shirou/gopsutil/internal/common"
+	"golang.org/x/sys/unix"
+)
+
+var ClocksPerSec = float64(128)
+var cpuMatch = regexp.MustCompile(`^CPU:`)
+var originMatch = regexp.MustCompile(`Origin\s*=\s*"(.+)"\s+Id\s*=\s*(.+)\s+Family\s*=\s*(.+)\s+Model\s*=\s*(.+)\s+Stepping\s*=\s*(.+)`)
+var featuresMatch = regexp.MustCompile(`Features=.+<(.+)>`)
+var featuresMatch2 = regexp.MustCompile(`Features2=[a-f\dx]+<(.+)>`)
+var cpuEnd = regexp.MustCompile(`^Trying to mount root`)
+var cpuCores = regexp.MustCompile(`FreeBSD/SMP: (\d*) package\(s\) x (\d*) core\(s\)`)
```

Comment:
sorry, I don't have a Dragonfly, so I can not check but is this `FreeBSD` correct ?

Reply:
Good catch. I just checked, and parsing the `dmesg` output is unlikely to work on FreeBSD, too, as regular users do not have access to `dmesg`. I'll remove this regexp because it will never match, and the correct value is eventually extracted in `InfoWithContext`.

Reply:
ok, I see why you parse the output of `dmesg`: to differentiate cpu vs. cores.

Reply:
Thank you for the information. I found it is not worked correctly in FreeBSD... it should be use `kern.sched.topology_spec` to getting how many psychical core it has. I will fix it, so we can left this at this time.

Reply:
I have rewritten the PR to use the equivalent `hw.cpu_topology.tree` in DragonflyBSD. Looking forward to seeing what you come up with on the FreeBSD side.